### PR TITLE
Added MongoDB Backend

### PR DESCRIPTION
**Pull request by nickGraham0**
See Teams shared folder for mongodb url needed in line 28 of index.js

### CODE REVIEW

**Comment by ikhyunAn** [comment]
RecipeBuilder/server/node_modules/mongoose/lib/drivers/node-mongodb-native/collection.js:185
          const err = new MongooseError(message);
                      ^

MongooseError: Operation `pantries.findOne()` buffering timed out after 10000ms
    at Timeout.<anonymous> (

File: server/index.js [hunk 1]
```diff
@@ -1,13 +1,36 @@
...
+         category: {type: String, required: false}
+      },
+   ],
+   current: {type: Boolean, default: false}
+});
+
+const Pantry = mongoose.model('Pantry', pantrySchema);
+
+
+mongoose.connect("MONGOBD_URL", { useNewUrlParser: true, useUnifiedTopology: true })
```
Comment (ikhyunAn) [inline server/index.js:28]: Is this where the provided MongoDB url go into?

Comment (ikhyunAn) [reply server/index.js:28]: Confirmed yes.

**Comment by nickGraham0** [comment]
Added all ip to whitelist. should resolve issue where other computers can't connect to db

File: server/index.js (hunk 1 above)
Comment (nickGraham0) [reply server/index.js:28]: Yes.

**Comment by ikhyunAn** [comment]
image
 Persistency confirmed

**Review by ikhyunAn — APPROVED** [review]
Confirmed Persistency through incognito mode

**Pull request merged**